tests/socket/socket_tests.cpp

```diff
@@ -48,20 +48,16 @@ connection_test(
     int local_address_length = 0;
     sender_socket.get_local_address(local_address, local_address_length);
 
-    connection_tuple_t tuple{};
+    connection_tuple_t tuple = {0};
     if (address_family == AF_INET) {
-        tuple.src_ip.ipv4 = htonl(INADDR_LOOPBACK);
         tuple.dst_ip.ipv4 = htonl(INADDR_LOOPBACK);
+        printf("tuple.dst_ip.ipv4 = %x\n", tuple.dst_ip.ipv4);
```

Comment:
was this a bug?

Reply:
no, earlier the test was configuring the map key with source IP also, but with connect redirect, the source IP will be all 0's so the key will not match anymore in the ebpf program, so the tests will fail.